service/history/execution/context.go

```diff
@@ -704,8 +720,8 @@ func (c *contextImpl) UpdateWorkflowExecutionTasks(
 		DomainName: domainName,
 	})
 	if err != nil {
-		if c.isPersistenceTimeoutError(err) {
-			c.notifyTasksFromWorkflowMutation(currentWorkflow, nil, true)
+		if isOperationPossiblySuccessfulError(err) {
+			notifyTasksFromWorkflowMutation(c.shard.GetEngine(), currentWorkflow, nil, true)
```

Comment:
this is not related to this PR but what happens if operation actually timed out and engine is still notified about those tasks that were not persisted?

Reply:
the tasks are possibly persisted, so we still notify the engine to try fetching those tasks